platform/commonUI/browse/res/templates/browse/object-properties.html

```diff
@@ -0,0 +1,60 @@
+<!--
+ Open MCT Web, Copyright (c) 2014-2015, United States Government
+ as represented by the Administrator of the National Aeronautics and Space
+ Administration. All rights reserved.
+
+ Open MCT Web is licensed under the Apache License, Version 2.0 (the
+ "License"); you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+ http://www.apache.org/licenses/LICENSE-2.0.
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ License for the specific language governing permissions and limitations
+ under the License.
+
+ Open MCT Web includes source code licensed under additional open source
+ licenses. See the Open Source Licenses file (LICENSES.md) included with
+ this source code distribution or the Licensing information page available
+ at runtime from the About dialog for additional information.
+-->
+<div ng-controller="ObjectInspectorController as controller">
+    <ul class="flex-elem grows l-inspector-part">
+        <li>
+            <em class="t-inspector-part-header">Properties</em>
+            <div class="inspector-properties"
+                 ng-repeat="data in metadata"
+                 ng-class="{ first:$index === 0 }">
+                <div class="label">{{ data.name }}</div>
+                <div class="value">{{ data.value }}</div>
+            </div>
+        </li>
+        <li ng-if="contextutalParents.length > 0">
+            <em class="t-inspector-part-header" title="The location of this linked object.">Location</em>
+            <span class="inspector-location"
+                  ng-repeat="parent in contextutalParents"
+                  ng-class="{ last:($index + 1) === contextualParents.length }">
+                <mct-representation key="'label'"
+                                    mct-object="parent"
+                                    ng-model="ngModel"
+                                    ng-click="ngModel.selectedObject = parent"
+                                    class="location-item">
+                </mct-representation>
+            </span>
+        </li>
+        <li ng-if="primaryParents.length > 0">
+            <em class="t-inspector-part-header" title="The location of the original object that this was linked from.">Original Location</em>
+            <span class="inspector-location"
+                  ng-repeat="parent in primaryParents"
+                  ng-class="{ last:($index + 1) === primaryParents.length }">
```

Comment:
Can’t remember if I added this or not. We can use CSS last-child instead.

From: Pete Richards <notifications@github.com<mailto:notifications@github.com>>
Reply-To: nasa/openmctweb <reply@reply.github.com<mailto:reply@reply.github.com>>
Date: Monday, February 8, 2016 at 12:57 PM
To: nasa/openmctweb <openmctweb@noreply.github.com<mailto:openmctweb@noreply.github.com>>
Cc: Charles Hacskaylo <charles.f.hacskaylo@nasa.gov<mailto:charles.f.hacskaylo@nasa.gov>>
Subject: Re: [openmctweb] [Plots] #638 New plot display options (#651)


In platform/commonUI/browse/res/templates/browse/object-properties.html<https://github.com/nasa/openmctweb/pull/651#discussion_r52214295>:

> +            <span class="inspector-location"
> +                  ng-repeat="parent in contextutalParents"
> +                  ng-class="{ last:($index + 1) === contextualParents.length }">
> +                <mct-representation key="'label'"
> +                                    mct-object="parent"
> +                                    ng-model="ngModel"
> +                                    ng-click="ngModel.selectedObject = parent"
> +                                    class="location-item">
> +                </mct-representation>
> +            </span>
> +        </li>
> +        <li ng-if="primaryParents.length > 0">
> +            <em class="t-inspector-part-header" title="The location of the original object that this was linked from.">Original Location</em>
> +            <span class="inspector-location"
> +                  ng-repeat="parent in primaryParents"
> +                  ng-class="{ last:($index + 1) === primaryParents.length }">


@charlesh88<https://github.com/charlesh88> @akhenry<https://github.com/akhenry> Any reason why we use a class for "last" instead of the last-child selector? Removing this would reduce watches.

If the last class is required, we should at the least use the $last boolean provided by ng-repeat.

—
Reply to this email directly or view it on GitHub<https://github.com/nasa/openmctweb/pull/651/files#r52214295>.